src/combat.cpp

```diff
@@ -961,12 +963,12 @@ void Combat::doAreaCombat(Creature* caster, const Position& position, const Area
 	leechCombat.origin = ORIGIN_NONE;
 	leechCombat.leeched = true;
 
-	for (Creature* creature : toDamageCreatures) {
+	for (const auto& creature : toDamageCreatures) {
 		CombatDamage damageCopy = damage; // we cannot avoid copying here, because we don't know if it's player combat
 		                                  // or not, so we can't modify the initial damage.
 		bool playerCombatReduced = false;
 		if ((damageCopy.primary.value < 0 || damageCopy.secondary.value < 0) && caster) {
-			Player* targetPlayer = creature->getPlayer();
+			auto targetPlayer = creature->getPlayer();
```

Comment:
Could this be `const auto`
 ```suggestion
			const auto targetPlayer = creature->getPlayer();
```